src/providers/chat_view_provider.ts

```diff
@@ -157,7 +157,11 @@ export class FlutterGPTViewProvider implements vscode.WebviewViewProvider {
             this._view?.webview.postMessage({ type: 'apiKeyValidation', value: 'Gemini API Key is valid' });
         } catch (error) {
             console.log('gemini api error', error);
-            this._view?.webview.postMessage({ type: 'apiKeyValidation', value: 'Gemini API Key is invalid' });
+            if (this.aiRepo?.isApiKeyInvalidError(error)) {
```

Comment:
Would it better to have this logic inside `validateApiKey` in aiRepo?

the `validateApiKey` should only throw error when the API key is invalid. This will reduce the statement to a single if else.